hibernate-core/src/main/java/org/hibernate/cfg/annotations/reflection/internal/JPAXMLOverriddenAnnotationReader.java

```diff
@@ -221,7 +221,7 @@ private enum PropertyType {
 	private static final Map<Class, String> annotationToXml;
 
 	static {
-		annotationToXml = new HashMap<>();
+		annotationToXml = new HashMap<>(78);
```

Comment:
When sizing with such precision please add a comment. Also try to consider how an HashMap is implemented: the size is an approximation that helps to compute a certain number of initial buckets, but the exact needed number will depend on the hash distribution;
 - underestimating might save a little memory (with luck) but might require an array resize to be triggered;
 - overestimating might end up having a couple of buckets which aren't strictly necessary

So hitting the exact number isn't very easy nor super important. If you want to really trim memory in such cases, such blocks of code might be better off by refactoring the code to remove the map altogether and think of something smarter to do - but I'm not sure if it's worth spending time on this particular map as there likely are more impactful changes you could make in the same time.